frida-gum/src/interceptor.rs

```diff
@@ -79,6 +79,66 @@ impl<'a> Interceptor<'a> {
         };
     }
 
+    /// Replace a function with another function. The new function should have the same signature
+    /// as the old one.
+    ///
+    /// # Safety
+    ///
+    /// Assumes that the provided function and replacement addresses are valid and point to the
+    /// start of valid functions
+    pub fn replace(
+        &mut self,
+        function: NativePointer,
+        replacement: NativePointer,
+        replacement_data: NativePointer,
+    ) -> Result<(), String> {
```

Comment:
It is generally bad form to return a string as an error instead of a specific error type. Don't worry about this for now, I'll deal with it afterwards to get the infrastructure in place.

Reply:
I know. I just wasn't sure how to go about setting up real error infrastructure.